### PR TITLE
Update cutter-deps to QT6

### LINKED ISSUE #3211: CMake >=2.27 - migrate from `FindPythonInterp` and `FindPythonLibs` to `FindPython`
These modules have been deprecated since CMake 3.12

```
CMake Warning (dev) at CMakeLists.txt:96 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found PythonInterp: /opt/homebrew/bin/python3.11 (found version "3.11.4")
CMake Warning (dev) at CMakeLists.txt:97 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found PythonLibs: /opt/homebrew/Frameworks/Python.framework/Versions/3.11/lib/libpython3.11.dylib (found suitable version "3.11.4", minimum required is "3.5")
CMake Warning at CMakeLists.txt:157 (message):
  KSyntaxHighlighting has been disabled because not supported in QT6
```
https://cmake.org/cmake/help/latest/policy/CMP0148.html

https://repology.org/project/cmake/versions

Currently the complete migration stopped by Ubuntu 18.04 - it has CMake [3.10.2](https://launchpad.net/ubuntu/+source/cmake). Bionic reached EOL already though, but still on extended support: https://endoflife.date/ubuntu

See also https://github.com/rizinorg/cutter/issues/2509

### PR DESCRIPTION
**Pull request by karliss**
IMPORTANT
**Do not merge** before   rizinorg/cutter-deps#16 is finished and  switching cutter-deps-qt URL back to rizinorg !
IMPORTANT

This is part 3/3 from
- rizinorg/cutter-deps-qt#7 
-  rizinorg/cutter-deps#16
-  rizinorg/cutter#3363

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

Update cutter deps to the new ones with QT6 and Python 3.12.

Reorganize Linux CI builds. There are now 4:
* linux-x86_64 -> building the main Appimage (using cutter-deps QT6) 
*  linux-x86_64-qt5  -> Appimage with maximum old system compatiblity using Qt5, should work on Ubuntu18.04 and similar age distros
* linux-x86_64-qt6-system-deps  -> system deps on a recent distro with Qt6
* linux-x86_64-system-deps -> system deps on oldest system (Ubuntu 18.04 + system Qt5)

The new linux Appimage should now include Qt wayland backend. 

macOS: 
Do both x86_64 and ARM64 builds on GHA. This somewhat makes the woodpecker build redundant. 
Do we still need it? If I remember correctly it was introduced when Github didn't have ARM64 builders yet. Currently not only github supports that but they are the default ones (macos-latest).
 
 
Meat of the testing should be done once the cutter-deps PR have been merged and this one is using using cutter-deps from them. I have done preliminary testing for most of the cases, but they should be redone with final build of cutter-deps once it's merged.

**Followup tasks**

* #3283 and
*  update the build documentation 

**Test plan (required)**

In each platform config make sure:
* Cutter opens
* Python sample plugin works
* Sample python plugin is properly listed in the plugin list

Testing platforms:
* [x] Windows
* [x] macOS arm native
* [ ] macOS intel (ideally tested on intel hardware instead of arm with Rosetta)
* [x] Linux qt6 appimage on Ubuntu 20.04
* [x] Linux qt6 appimage on a more recent distro
    * [x] with desktop in wayland mode
    * [x] with desktop in x11 mode
* [x] Linux qt5 appimage on Ubuntu 18.04

**Closing issues**

closes #3211
unblocks #3296

**TODO**
* [x] update the example python plugin
* [x] appveyor build
* [x] cleanup commented out x86_64-qt5 x86_64-system-deps configs, which are replaced by seperate jobs matrix (due to how docker in GHA works)

### CODE REVIEW

File: scripts/fetch_deps.sh [hunk 1]
```diff
@@ -5,19 +5,27 @@ set -e
 cd $(dirname "${BASH_SOURCE[0]}")/..
 mkdir -p cutter-deps && cd cutter-deps
 
-DEPS_FILE_linux_x86_64=cutter-deps-linux-x86_64.tar.gz
-DEPS_SHA256_linux_x86_64=0721c85548bbcf31f6911cdb2227e5efb4a20c34262672d4cd2193db166b2f8c
+DEPS_BASE_URL=https://github.com/karliss/cutter-deps/releases/download/qt6-test #TODO: replace before merging
```
Comment (karliss) [inline scripts/fetch_deps.sh]: Switch back before merging the PR.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -29,19 +29,13 @@ jobs:
       matrix:
         name: [
             linux-x86_64,
-            linux-x86_64-system-deps,
             linux-x86_64-qt6-system-deps,
```
Comment (XVilka) [inline .github/workflows/ci.yml]: Consider renaming `linux-x86_64-qt6-system-deps` into just `linux-x86_64-system-deps` and the old one just into `linux-x86_64-qt5-system-deps` to reflect the fact that Qt6 is the default one now.

File: src/bindings/bindings.xml.in [hunk 3]
```diff
@@ -31,9 +31,19 @@
         PyErr_Print();
         return QString();
         }
-        PythonToCppFunc pythonToCpp = Shiboken::Conversions::isPythonToCppConvertible(Sbk${PYSIDE_NAME}_QtCoreTypeConverters[SBK_QSTRING_IDX], pyResult);
+        /*PythonToCppFunc pythonToCpp = Shiboken::Conversions::isPythonToCppConvertible(Sbk${PYSIDE_NAME}_QtCoreTypeConverters[SBK_QSTRING_IDX], pyResult);
```
Comment (XVilka) [inline src/bindings/bindings.xml.in]: Why comment and not remove instead?

**Review by XVilka — CHANGES_REQUESTED** [review]

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
Tested it (basic testing procedure) on these platforms:
- macOS 14.5 ARM64
- Windows 11 (x86_64)
- also a friend of mine checked on macOS x86_64 natively - seems everything is in order.

**Comment by Rot127** [comment]
Both AppImages work for me (qt5 and qt6). Tested on Debian 11 x86_64

```
echo $XDG_SESSION_TYPE
x11
```

**Comment by wargio** [comment]
i have tested linux qt5 and qt6 and works.
```
$ echo $XDG_SESSION_TYPE
wayland
```

**Comment by insaneninja117** [comment]
This is probably obvious to everyone already but unless a special Qt5 build is added for Windows moving forwards this will mean support for Windows 7 will be dropped with this change.

**Comment by XVilka** [comment]
> This is probably obvious to everyone already but unless a special Qt5 build is added for Windows moving forwards this will mean support for Windows 7 will be dropped with this change.

**Comment by XVilka** [comment]
Sorry, pressed the wrong button

File: scripts/fetch_deps.sh (hunk 1 above)
Comment (Rot127) [reply scripts/fetch_deps.sh]: Reminder :)

**Review by Rot127 — APPROVED** [review]
Only revert the link to the Rizin repo please.

**Review by wargio — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Pull request merged**